zero_bin/rpc/src/jumpdest.rs

```diff
@@ -0,0 +1,368 @@
+                ensure!(
+                    call_stack.is_empty().not(),
+                    "Call stack was empty when a JUMPDEST was encountered."
+                );
+                let (code_hash, ctx) = call_stack.last().unwrap();
+                let jumped_here = if let Some(jmp_target) = prev_jump {
+                    ensure!(
+                        jmp_target == entry.pc,
+                        "The structlog seems to make improper JUMPs."
+                    );
+                    true
+                } else {
+                    false
+                };
+                let jumpdest_offset = entry.pc as usize;
+                if jumped_here {
+                    jumpdest_table.insert(code_hash, *ctx, jumpdest_offset);
+                }
+                // else: we do not care about JUMPDESTs reached through fall-through.
+                prev_jump = None;
+            }
+            "EXTCODECOPY" | "EXTCODESIZE" => {
+                next_ctx_available += 1;
+                prev_jump = None;
+            }
+            "RETURN" | "REVERT" | "STOP" => {
+                ensure!(call_stack.is_empty().not(), "Call stack was empty at POP.");
+                call_stack.pop().unwrap();
+                prev_jump = None;
+            }
+            _ => {
+                prev_jump = None;
+            }
+        }
+
+        prev_depth = curr_depth;
+        prev_op = op;
+    }
+    Ok(jumpdest_table)
+}
+
+pub mod structlogprime {
+    use std::collections::BTreeMap;
+
+    use alloy::rpc::types::trace::geth::DefaultFrame;
+    use alloy_primitives::{Bytes, B256, U256};
+    use serde::{ser::SerializeMap as _, Deserialize, Serialize, Serializer};
+    use serde_json::Value;
+
+    /// Geth Default struct log trace frame
+    ///
+    /// <https://github.com/ethereum/go-ethereum/blob/a9ef135e2dd53682d106c6a2aede9187026cc1de/eth/tracers/logger/logger.go#L406-L411>
+    #[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub(crate) struct DefaultFramePrime {
+        /// Whether the transaction failed
+        pub failed: bool,
+        /// How much gas was used.
+        pub gas: u64,
+        /// Output of the transaction
+        #[serde(serialize_with = "alloy_serde::serialize_hex_string_no_prefix")]
+        pub return_value: Bytes,
+        /// Recorded traces of the transaction
+        pub struct_logs: Vec<StructLogPrime>,
+    }
+
+    /// Represents a struct log entry in a trace
+    ///
+    /// <https://github.com/ethereum/go-ethereum/blob/366d2169fbc0e0f803b68c042b77b6b480836dbc/eth/tracers/logger/logger.go#L413-L426>
+    #[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+    pub(crate) struct StructLogPrime {
+        /// program counter
+        pub pc: u64,
+        /// opcode to be executed
+        pub op: String,
+        /// remaining gas
+        pub gas: u64,
+        /// cost for executing op
```

Comment:
nit: consistency with below

```suggestion
        /// Program counter
        pub pc: u64,
        /// Opcode to be executed
        pub op: String,
        /// Remaining gas
        pub gas: u64,
        /// cost for executing op
```

Reply:
Thanks.